audit.go

```diff
@@ -615,7 +615,7 @@ func (s AuditStatus) toWireFormat() []byte {
 // struct.
 func (s *AuditStatus) FromWireFormat(buf []byte) error {
 	if len(buf) < sizeofAuditStatus {
-		return io.ErrUnexpectedEOF
```

Comment:
If I understand correctly this bug was introduced by #93, which grew the size of the AuditStatus struct by an extra uint32. I'm worried blindly accepting any packet length can cause issues in the future. WDYT about checking for at least the previously expected length?

Reply:
That was my initial sketch, but then I saw the comment that I missed taking note of in #93 that said that we won't look at the length at all. I'd be happy to go with that. This is what I had prepared.

```
const (
	sizeofAuditStatus = int(unsafe.Sizeof(AuditStatus{}))

	// MinSizeofAuditStatus is the minimum usable message size that
	// is acceptable for unmarshaling from the wire format. Messages
	// this size do not report backlog_wait_time_actual.
	MinSizeofAuditStatus = sizeofAuditStatus - int(unsafe.Sizeof(uint32(0)))
)

func (s AuditStatus) toWireFormat() []byte {
	return (*[sizeofAuditStatus]byte)(unsafe.Pointer(&s))[:]
}

// FromWireFormat unmarshals the given buffer to an AuditStatus object.
// It returns an io.ErrUnexpectedEOF if len(buf) is less than MinSizeofAuditStatus.
func (s *AuditStatus) FromWireFormat(buf []byte) error {
	if len(buf) < MinSizeofAuditStatus {
		return io.ErrUnexpectedEOF
	}
	copy((*[unsafe.Sizeof(AuditStatus{})]byte)(unsafe.Pointer(s))[:], buf)
	return nil
}
```

Another alternative is to still unmarshal the data, and return the error if it's too short, with a new `error` type that says what fields exist.

Which would you prefer?

Reply:
> This is what I had prepared.

It looks perfectly fine to me. I'd say we go with this.

> Another alternative is to still unmarshal the data, and return the error if it's too short, with a new error type that says what fields exist.

Sounds overly complicated. Will it be returning an error in kernels without `backlog_wait_time_actual` and expecting the receiver to have special handling for this "error"? We can discuss separately how to best propagate to the user which fields are present in the status struct, if we want to do so, but in the context of a bugfix I'll say let's just restore the original behavior.